source/AI.cpp

```diff
@@ -1810,11 +1819,31 @@ bool AI::MoveTo(Ship &ship, Command &command, const Point &targetPosition, const
 	bool shouldReverse = false;
 	dp = targetPosition - StoppingPoint(ship, targetVelocity, shouldReverse);
 
-	bool isFacing = (dp.Unit().Dot(angle.Unit()) > .95);
+	// Calculate target vector required to get where we want to be.
+	Point tv = dp;
```

Comment:
Should we be adjusting the "stopping point" instead of calculating where we want to be "twice"?

Reply:
The "stopping point" is still the same before and after this PR; the only difference is that we now also check at what speed we want to get there (instead of always going full-speed).
In the situation where a ship is moving towards its target, but at a higher speed than the cruiseSpeed, then we would want the ship to actively slow down; so the calculation of the desired speed can result in a vector opposite of the vector directly towards the target to ensure the ship slows down.
Maybe I'm missing something, but I don't see which part of the calculation can be optimized to avoid a double calculation.